### ISSUE TITLE
Error pushing Node 6.11.1 app to cloud foundry using buildpack 1.6.2

### ISSUE THREAD

**Issue by unochild** [cloudfoundry/nodejs-buildpack#98]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?

CF: 2.61.0
CF CLI: 6.22.2-nimbus-e480663-2016-11-15

What version of the buildpack you are using?

1.6.2

If you were attempting to accomplish a task, what was it you were attempting to do?

Push a node 6.11.1 app to cloud foundry

What did you expect to happen?

Deploy successfully

What was the actual behavior?

Threw the following error

```-----> Downloaded app package (6.7M)
Cloning into '/tmp/buildpacks/buildpack-nodejs'...
Note: checking out 'b24343e5a62ee977e6f82b540640a9a0d818bdf7'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:
  git checkout -b new_branch_name
-----> Download go 1.8.1
-----> Running go build supply
-----> Nodejs Buildpack version 1.6.2
-----> Installing binaries
       engines.node (package.json): 6.11.1
       engines.npm (package.json): unspecified (use default)
-----> Installing node 6.11.1
       Download [https://buildpacks.cloudfoundry.org/dependencies/node/node-6.11.1-linux-x64-19c09bc7.tgz]
       Using default npm version: 3.10.10
-----> Installing yarn 0.27.5
       Download [https://buildpacks.cloudfoundry.org/dependencies/yarn/yarn-v0.27.5-73002997.tar.gz]
       Installed yarn 0.27.5
-----> Creating runtime environment
-----> Running go build finalize
       PRO TIP: It is recommended to vendor the application's Node.js dependencies
       Visit http://docs.cloudfoundry.org/buildpacks/node/index.html#vendoring
       NODE_ENV=production
       NODE_TLS_REJECT_UNAUTHORIZED=0
       NODE_HOME=/tmp/staged/app/.cloudfoundry/0/node
       NODE_MODULES_CACHE=true
       NODE_VERBOSE=false
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
-----> Restoring cache
       Skipping cache restore (no previous cache)
-----> Building dependencies
       Running bash build/setupNimbusEnv.sh (npm)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6676d5]
goroutine 1 [running]:
nodejs/finalize.(*Finalizer).runScript(0xc4200b5ec0, 0xc420119800, 0x1c, 0x6fe1cb, 0x3, 0x0, 0x0)
	/tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:345 +0x235
nodejs/finalize.(*Finalizer).runPrebuild(0xc4200b5ec0, 0x6fe1cb, 0x3, 0x0, 0x0)
	/tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:326 +0x76
nodejs/finalize.(*Finalizer).BuildDependencies(0xc4200b5ec0, 0x7100b8, 0xc4200b5ec0)
	/tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:206 +0xac
nodejs/finalize.Run(0xc4200b5ec0, 0x0, 0x0)
	/tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:89 +0x191
main.main()
	/tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/cli/main.go:69 +0x706
Staging failed: Buildpack compilation step failed

FAILED
Error restarting application: BuildpackCompileFailed```

Please confirm where necessary:
* [x] I have included a log output
* [x] My log includes an error message
* [x] I have included steps for reproduction

**Comment by sclevine** [comment]
@unochild Can you provide a sample app that reproduces this failure?

Also, can you confirm that the latest version of the buildpack (1.6.4) reproduces this failure?

**Comment by DominicGinger** [comment]
Same issue here. It happens on any node project with 6.11.1 and a `heroku-postbuild` or `heroku-prebuild` npm script. Buildpack 1.5.* works, but 1.6+ (including 1.6.4) produces the error above.

**Comment by DominicGinger** [comment]
Minimum requirements to replicate:
### Package.json
```javascript
{
  "name": "test_buildpack",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "engines": {
    "node": "6.11.1"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "heroku-prebuild": "echo \"Hello world\""
  },
  "author": "",
  "license": "UNLICENSED"
}
```

### manifest.yml
```yaml
---
applications:
  - memory: 256M
    buildpack: https://github.com/cloudfoundry/buildpack-nodejs#v1.6.4
    stack: cflinuxfs2
```

### Log output
```sh
-----> Downloaded app package (4.0K)
-----> Downloaded app buildpack cache (4.0K)
Cloning into '/tmp/buildpacks/buildpack-nodejs'...
Note: checking out 'a6f97c9581f2bca2111644dcae15d9d47e74b8e8'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:
  git checkout -b new_branch_name
-----> Download go 1.8.1
-----> Running go build supply
-----> Nodejs Buildpack version 1.6.4
-----> Installing binaries
       engines.node (package.json): 6.11.1
       engines.npm (package.json): unspecified (use default)
-----> Installing node 6.11.1
       Download [https://buildpacks.cloudfoundry.org/dependencies/node/node-6.11.1-linux-x64-19c09bc7.tgz]
       Using default npm version: 3.10.10
-----> Installing yarn 0.27.5
       Download [https://buildpacks.cloudfoundry.org/dependencies/yarn/yarn-v0.27.5-73002997.tar.gz]
       Installed yarn 0.27.5
-----> Creating runtime environment
-----> Running go build finalize
       PRO TIP: It is recommended to vendor the application's Node.js dependencies
       Visit http://docs.cloudfoundry.org/buildpacks/node/index.html#vendoring
       NODE_ENV=production
       NODE_HOME=/tmp/staged/app/.cloudfoundry/0/node
       NODE_MODULES_CACHE=true
       NODE_VERBOSE=false
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
-----> Restoring cache
       Running echo "Hello world" (npm)
       Skipping cache restore (new runtime signature)
-----> Building dependencies
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x667e05]
goroutine 1 [running]:
nodejs/finalize.(*Finalizer).runScript(0xc4200bfec0, 0xc42012d3a0, 0x12, 0x6ff3eb, 0x3, 0x0, 0x0)
        /tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:345 +0x235
nodejs/finalize.(*Finalizer).runPrebuild(0xc4200bfec0, 0x6ff3eb, 0x3, 0x0, 0x0)
        /tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:326 +0x76
nodejs/finalize.(*Finalizer).BuildDependencies(0xc4200bfec0, 0x711330, 0xc4200bfec0)
        /tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:206 +0xac
nodejs/finalize.Run(0xc4200bfec0, 0x0, 0x0)
        /tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/finalize.go:89 +0x191
main.main()
        /tmp/buildpacks/buildpack-nodejs/src/nodejs/finalize/cli/main.go:69 +0x706
Staging failed: Buildpack compilation step failed

FAILED
Error restarting application: BuildpackCompileFailed

TIP: use 'cf logs broken_buildpack --recent' for more information

```

**Comment by rportugal** [comment]
#99 fixes the crash, however it doesn't seem to do anything.

**Comment by sclevine** [comment]
Confirmed. We'll fix it and cut a release ASAP.

**Comment by rportugal** [comment]
I've just tested with 1.6.5 and it seems to be working. @sclevine can you confirm it was fixed?

**Comment by sclevine** [comment]
Yes, this is fixed in 1.6.5. This issue was left out of the release notes by mistake. Sorry for the confusion. We'll add this to the release notes soon.
